docker/docker.go

```diff
@@ -634,16 +638,53 @@ func getRegistryAuth(pr v3.PrivateRegistry) (string, error) {
 }
 
 func GetImageRegistryConfig(image string, prsMap map[string]v3.PrivateRegistry) (string, string, error) {
+	/*
+		Image can be passed as
+		- repo.com/foo/bar/rancher/rke-tools:v0.1.51
+		or
+		- repo.com/rancher/rke-tools:v0.1.51
+		or
+		- rancher/rke-tools
+		Where the repo can be:
+		- repo.com
+		or
+		- repo.com/foo/bar
+		When checking for the repo presence in prsMap, for the example above following registries will be tried
+		in the order:
+		- repo.com/foo/bar
+		- repo.com/foo
+		- repo.com
+	*/
 	namedImage, err := ref.ParseNormalizedNamed(image)
 	if err != nil {
 		return "", "", err
 	}
+	if len(prsMap) == 0 {
+		return "", "", nil
+	}
 	regURL := ref.Domain(namedImage)
-	if pr, ok := prsMap[regURL]; ok {
-		// We do this if we have some docker.io login information
-		regAuth, err := getRegistryAuth(pr)
-		return regAuth, pr.URL, err
+	regPath := ref.Path(namedImage)
+	splitPath := strings.Split(regPath, "/")
```

Comment:
If the user has provided `repo.com/foo/bar` as the registry URL, why do we have to check repo.com/foo and repo.com? That seems like an unintended side/extra behavior and we would have to document this behavior if we want to provide this functionality. If the user intends to use these different search paths, it's better to have the same explicitly defined in the config.

Reply:
@leodotcloud the logic works this way:

* system images can be coming from diff repos (although rarely used, we've always had support for it in rke)
* For every image, we do the repo lookup. If we are to add search path explicitly, you'd have to do it for each and every image, and that's more cumbersome in my opinion.